### PR TITLE
Added Decision tree model

### PR DESCRIPTION
**Pull request by Sar-thak-3**

### CODE REVIEW

File: src/slowmokit/methods/neighbors/decision_tree/decision_tree.cpp [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+        std::vector<std::vector<double>> left,right;
+        std::vector<std::vector<std::vector<double>>> temp = divideData(xData,fkey,fval);
+        left = temp[0];
+        right = temp[1];
+        double l = left.size()/double(xData.size());
+        double r = right.size()/double(xData.size());
+        if(left.size()==0 or right.size()==0){
+            return -1;
+        }
+        std::vector<double> y_data,lY,rY;
```
Comment (ken1000minus7) [inline src/slowmokit/methods/neighbors/decision_tree/decision_tree.cpp:60]: Fix the case of `y_data`

File: src/slowmokit/methods/neighbors/decision_tree/decision_tree.hpp [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+     * @param minSamplesL minimum samples to be in leaf node by default 1
+     * @param max_f maximum samples to be in leaf nodes
+     */
+    DecisionTree(int max_d,int minSamplesL,int max_f);
+    /**
+     * @brief prepare decision tree
+     * 
+     * @param x_data all training values
+     */
+    void train(std::vector<std::vector<double>> x_data);
```
Comment (ken1000minus7) [inline src/slowmokit/methods/neighbors/decision_tree/decision_tree.hpp:65]: Also fix the cases here

**Review by ken1000minus7 — CHANGES_REQUESTED** [review]
Just small changes needed

**Comment by uttammittal02** [comment]
Need much improvements yet

**Pull request closed without merging**